Updated packages and fixed the typescript compile error

nextsummer33 opened the pull request:
Most of packages has been updated and changed the configuration of typescript compiler `tsconfig.js` to remove compile errors and warnings. Now it can be run smoothly via dotnet run

Commit:
[*] projects, renamed npm-shrinkwrap.json to project-lock.json. Remove the warning for npm install
[*] tsconfig.json, Added types attribute. changed "strict" attribute to "false" since this will cause typescript compiler to throw an error about class component decorator.
[*] packages, updated most of packages

Commit:
[+] project, added babel transform runtime and eslint supported. Added a editor config file and make the tab indent more consistent
[*] code, change " -> ' for javascript file
[*] folder, move test spec files into specs folder
[*] packages, updated mocha and sinon version to newest

Commit:
[+] gitignore, ignored .DS_Store

Commit:
[+] eslint, added support eslint for vue file
[*] code, cleaned up code to follow the rule of eslint
[*] test, moved karma.conf.js into test/unit folder. since it only used for testing

Commit:
[v] components/fetchdata, added bind key to vue for loop
[*] packages, run eslint checking before running the test

Commit:
[-] project, removed package-lock.json
[+] gitignore, added a ignore for package-lock.json

Commit:
[-] components, removed components inside the apps [too much]
[+] router, separated into other folder and added hooks for components to access a callback when router-view is changing
[+] vuex, added vuex-typescript supported
[*] fetchdata, applied vuex-typescript into the component.
[*] folders, rename css folder to static, this folder should use to contain images and css etc.

Commit:
[+] vuex, added a simple decorator to map the getter function of a store.
[+] vuex, added a state.ts  to declare vuex store state
[*] project, changed the name of vue-shim.d.ts file to typings.d.ts and moved into ClientApp folder
[*] components, added a store control to the sample fetchdata component to demonstrate how to use vuex-typescript

Commit:
[+] test, added a vuex module test
[+] packages, added sinon-stub-promise for testing
[*] packages, updated sinon version to v4.1.1
[*] fetchdata, applied async and await to dispatch action to store

raghur commented:
Thanks for the PR - looks interesting. One question - shouldn't the lock file be package-lock.json instead of project-lock.json?

Commit:
[+] Project/Package, added missing eslint-vue-plugin.
[+] Project/Webpack, added babel-polyfill in order to support ie 11.
[*] Client/Store, changed dispatch actions to async syntax.
[*] Project/Package, updated all packages.
[v] Project/Eslint, fixed the settings for eslint-vue-plugin.
[v] Project/Webpack, fixed the vendor js under uglifying problem.
[v] Client/Components, fixed the template to allow to use vue-loader version 13.0.0+ since 13.0.0 has a breaking changes for require('comp') -> require('comp').default.

Ku4nCheang commented:
I just followed the recommendation messages under using npm install. I did not notice the npm-shrinkwrap was used for supporting older version of npm.

Commit:
[-] Project, removed project-lock.json since we don't care older version of npm. ;) This is a template that uses newest packages.
